components/builder-jobsrv/src/server/log_archiver/s3.rs

```diff
@@ -115,40 +84,19 @@ impl S3Archiver {
 impl LogArchiver for S3Archiver {
     fn archive(&self, job_id: u64, file_path: &PathBuf) -> Result<()> {
         let mut buffer = Vec::new();
-        let mut put_object = PutObjectRequest::default();
-        put_object.bucket = self.bucket.clone();
-        put_object.key = Self::key(job_id);
+        let mut request = PutObjectRequest::default();
+        request.bucket = self.bucket.clone();
+        request.key = Self::key(job_id);
 
         let mut file = OpenOptions::new().read(true).open(file_path)?;
         file.read_to_end(&mut buffer)?;
-        put_object.body = Some(buffer.as_slice());
-
-        // This panics if it can't resolve the URL (e.g.,
-        // there's a netsplit, your Minio goes down, S3 goes down (!)).
-        // We have to catch it, otherwise no more logs get captured!
-        //
-        // The code in the S3 library we're currently using isn't
-        // UnwindSafe, so we need to deal with that, too.
-        let result = panic::catch_unwind(AssertUnwindSafe(|| {
-            self.client.put_object(&put_object, None)
-        }));
-
-        match result {
-            Ok(Ok(_)) => Ok(()), // normal result
-            Ok(Err(e)) => {
-                // This is a "normal", non-panicking error, e.g.,
-                // they're configured with a non-existent bucket.
-                Err(Error::JobLogArchive(job_id, e))
-            }
+        request.body = Some(buffer.as_slice().to_vec());
+
+        match self.client.put_object(&request).sync() {
+            Ok(_) => Ok(()),
```

Comment:
Is it worth adding some debug logging here for the success case?

Reply:
I just left the logging mostly as I found it for now..